common/log/zap_logger_test.go

```diff
@@ -188,3 +191,55 @@ func TestEmptyMsg(t *testing.T) {
 	fmt.Println(out, lineNum)
 	assert.Equal(t, `{"level":"info","msg":"`+defaultMsgForEmpty+`","error":"test error","wf-action":"add-workflow-started-event","logging-call-at":"zap_logger_test.go:`+lineNum+`"}`+"\n", out)
 }
+
+func TestRotationLogger(t *testing.T) {
+	dir := testutils.MkdirTemp(t, "", "testRotationLogger")
+	cfg := Config{
+		Level:          "info",
+		OutputFile:     dir + "/test.log",
+		EnableRotation: true,
+		MaxSize:        1,
+	}
+
+	log, err := BuildZapLogger(cfg)
+	assert.NoError(t, err)
+	assert.NotNil(t, log)
+	for i := 0; i < 20000; i++ {
+		log.Info("this is dump log")
+	}
+	files, err := os.ReadDir(dir)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(files))
+	sort.Slice(files, func(i, j int) bool {
+		fileI, _ := files[i].Info()
+		fileJ, _ := files[j].Info()
+		return fileI.Size() < fileJ.Size()
+	})
+	rotatedFile, _ := files[1].Info()
+	currentFile, _ := files[0].Info()
+	assert.LessOrEqual(t, int64(1024*1024)-rotatedFile.Size(), int64(100))
+	assert.Less(t, currentFile.Size(), int64(1024*1024))
+}
+
+func TestNoRotationLogger(t *testing.T) {
```

Comment:
Nit: how about
```suggestion
func TestLogger_FilesArentRotated_WhenDisabled(t *testing.T) {
```

That's cleared about what you're testing (and the above could be `TestLogger_FilesAreRotated_WhenEnabled` or something)